### PR TITLE
Multi-Step Form + Bugfix Style Select JavaScript 

### PR DESCRIPTION
**Pull request by wisam-alhennawi**
- Multi-Step form: a new feature has been added to build a multi step form.
- Bugfix: styled select component (Javascript error if it used more than one times in the same form)

### CODE REVIEW

**Review by jpbu — COMMENTED** [review]
Thanks, I reviewed the code changes and tested both a normal tab and multi-step locally with a test app, worked well.

In addition to the inline comments, the following points:
* A description in the README ideally with an example would be very helpful
* It looks to me like the `MultiStepForm` trait only works if the `HasTabs` trait is used, so I would rather add the `use MultiStepForm` to the `HasTabs` than the general `LaraFormsBuilder`
* Styling of tabs in the multi-step case should be changed to not imply that the tabs are clickable/navigateable
* With `confirmBeforeSubmit` set to true, "Previous step" navigation on the confirmation view should reset the confirm mode back to an editable view (in my test, it was just showing read only and there was no way to edit it)

File: src/LaraFormsBuilder.php [hunk 1]
```diff
@@ -349,17 +360,21 @@ public function switchToEditMode()
...
-        $customMessageKey = 'A new '.$modelName.' has been created successfully.';
-        $message = trans('A new entry has been created successfully.');
-        if ($this->mode == 'update') {
-            $customMessageKey = 'The '.$modelName.' has been updated successfully.';
-            $message = trans('Changes were saved successfully.');
-        }
+        if ($this->customSuccessMessage != '') {
+            $message = $this->customSuccessMessage;
+        } else {
+            // if you want to customize the success message, you should add the custom message key entry to the lang file (Example: "A new forestryPoolMember has benn created successfully.") or override the method
```
[the remark this answers is not in the document]
Comment (wisam-alhennawi) [reply src/LaraFormsBuilder.php:366]: You're right, I'll add it.

**Comment by wisam-alhennawi** [comment]
> Thanks, I reviewed the code changes and tested both a normal tab and multi-step locally with a test app, worked well.
> 
> In addition to the inline comments, the following points:
> 
> * A description in the README ideally with an example would be very helpful
> * It looks to me like the `MultiStepForm` trait only works if the `HasTabs` trait is used, so I would rather add the `use MultiStepForm` to the `HasTabs` than the general `LaraFormsBuilder`
> * Styling of tabs in the multi-step case should be changed to not imply that the tabs are clickable/navigateable
> * With `confirmBeforeSubmit` set to true, "Previous step" navigation on the confirmation view should reset the confirm mode back to an editable view (in my test, it was just showing read only and there was no way to edit it)

Thanks for your review and comments. 
Regarding a description in the README, you are right and it is already in my ToDos but I didn't have time to do it yesterday :).
I will add it and implement the other points.

**Comment by wisam-alhennawi** [comment]
@jpbu Thank you again for your comments, I implemented them. 
> Styling of tabs in the multi-step case should be changed to not imply that the tabs are clickable/navigateable

In this context I only distinguished the navigation part. We can also split` tabs.blade.php` into two files (tabs and steps) if needed in the future

**Review by jpbu — APPROVED** [review]
Thanks, I double checked the code changes and tested locally with a test app, looks good. Ready to merge from my perspective.

**Pull request merged**